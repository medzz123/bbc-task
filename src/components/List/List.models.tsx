export interface ListProps {
  type?: 'ordered' | 'unordered';
  items: string[];
}
