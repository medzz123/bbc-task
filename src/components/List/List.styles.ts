import { styled } from '@theme/config';

export const ListContainer = styled('ul', {
  padding: '0',
  margin: '0',
  li: {
    listStylePosition: 'inside',
  },
  color: '$text',
});
