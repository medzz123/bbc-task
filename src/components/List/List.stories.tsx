import { Meta, Story } from '@storybook/react/types-6-0';
import React from 'react';

import List from './List';

export default {
  title: 'Components/List',
  component: List,
} as Meta;

const Template: Story = (storyArguments) => <List {...storyArguments} />;

export const Default = Template.bind({});
