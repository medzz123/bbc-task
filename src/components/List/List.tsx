import { hashString } from '@lib/hashString';
import React, { FunctionComponent } from 'react';

import { ListProps } from './List.models';
import { ListContainer } from './List.styles';

const List: FunctionComponent<ListProps> = (props) => {
  const { type = 'unordered', items } = props;

  return (
    <ListContainer as={type === 'unordered' ? 'ul' : 'ol'} data-testid="list">
      {items.map((listItem) => (
        <li key={hashString(listItem)}>{listItem}</li>
      ))}
    </ListContainer>
  );
};

export default List;
