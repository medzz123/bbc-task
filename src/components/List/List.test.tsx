import { render } from '@test/library';
import React from 'react';

import List from './List';

test('List Test', () => {
  const { container } = render(<List items={[]} />);

  expect(container).toMatchSnapshot();
});
