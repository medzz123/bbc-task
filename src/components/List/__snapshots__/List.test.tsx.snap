// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Test 1`] = `
<div>
  <ul
    class="sx7p6k5"
    data-testid="list"
  />
</div>
`;
